pipenv/cli/command.py

```diff
@@ -714,17 +714,22 @@ def verify(state):
     context_settings=CONTEXT_SETTINGS,
 )
 @option("--dev", is_flag=True, default=False, help="Also add development requirements.")
+@option("--dev-only", is_flag=True, default=False, help="Only add development requirements.")
+@option("--hash", is_flag=True, default=False, help="Add package hashes.")
 @pass_state
-def requirements(state, dev=False):
+def requirements(state, dev=False, dev_only=False, hash=False):
     lockfile = state.project.lockfile_content
     for i, package_index in enumerate(lockfile['_meta']['sources']):
         prefix = '-i' if i == 0 else '--extra-index-url'
         echo(crayons.normal(' '.join([prefix, package_index['url']])))
-    for req_name, value in lockfile['default'].items():
-        echo(crayons.normal(f"{req_name}{value['version']}"))
-    if dev:
+    if not dev_only:
+        for req_name, value in lockfile['default'].items():
+            hashes = [f' \\\n    --hash={h}' for h in value['hashes'] if hash]
+            echo(crayons.normal(''.join([req_name, value['version'], *hashes])))
+    if dev or dev_only:
         for req_name, value in lockfile['develop'].items():
-            echo(crayons.normal(f"{req_name}{value['version']}"))
+            hashes = [f' \\\n    --hash={h}' for h in value['hashes'] if hash]
```

Comment:
I might be wrong here, but it seems to me you are checking if hash for every `value['hashes']`.
Can you rewrite this as:
```
if hash:
    hashes = [f' \\\n    --hash={h}' for h in value['hashes'] if hash]
else:
    hashes = []
```

Reply:
Fixed